# Task1

## Lady Tasting Tea

An event in Fisher's life led him to design this experiment, His friend's future wife, Muriel Bristol, said that she could tell if the tea or the milk was put first into a cup. She was given eight cups, four of them created each way and given to her in random order. Fisher wanted to find the probability that she could say the order of addition to the cup. In the end, Muriel correctly identified which of the four cups of tea had milk poured before the tea.

Problem Statement
Suppose the Lady Tasting Tea experiment is altered to involve twelve cups of tea. Six have the milk in first and the other six having tea in first. A person claims that they have the special power of being able to tell if the tea or the milk went into a cup first upon tasting it. You agree to accept their claim if they can tell which of the six cups in your experiment had the milk in first.
1.	Calculate, using Python, the probability that they select the correct six cups.
Suppose, now, you are willing to accept one error. Once they select the six cups they think had the milk in first, you will give them the benefit of the doubt should they have selected at least five of the correct cups.
2.	Calculate the probability, assuming they have no special powers, that the person makes at most one error.
3.	Would you accept two errors? Explain.



In [1]:
# Mathematical functions from the standard library.
# https://docs.python.org/3/library/math.html
import math

For this task, we will be using 12 cups of tea as the test. 

In [2]:
# total number of cups.
cups = 12

# Milk first.
milk_first = 6

# Tea first.
tea_first = 6

We can show that from 12 cups, there is a possibility of choosing 1 from 12 choices, that leaves 11 cups to choose from. 1 out of 11 leaves 10 and so on.
So, there are 12 x 11 10 x 9 x 8 x 7 choices of cup as the test progresses. After 6 cups have been selected the rest are ignored. 

In [3]:
# Total ways of selecting 6 random cups.
selection = 12 * 11 * 10 * 9 * 8 * 7

# Display total number of possible selections.
selection

665280

Now that we know how many permutations there are of selecting 6 cups from 12, we want to see the number ways we can randomise the presentation of the cups to the tester

In [4]:
# How many chances are there that the lady will select correctly out of the 6 cups.
random = 6 * 5 * 4 * 3 * 2 * 1

# Display total number of possible selections.
random

720

In [5]:
# No of permutations that the tester can decide which liquid came first.
permutations = selection // random

# Show.
permutations

924

Therefore, we can see that the chance of selecting the six cups with milk poured first from 12 cups is 1 in 924.
   Or 1/924 gives us a probability of 0.001 (rounded down).